xarray/core/groupby.py

```diff
@@ -429,6 +430,8 @@ def _parse_group_and_groupers(
             )
             for group, grouper in grouper_mapping.items()
         )
+        if isinstance(group, str):
```

Comment:
instead I'd return a flag `squeeze_group_label_on_iter` that is True if group is `str`